### PR TITLE
WT-4972 Run unit tests on red hat

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -1266,6 +1266,74 @@ buildvariants:
     - name: unit-test-bucket06
     - name: unit-test-bucket07
 
+- name: rhel80
+  display_name: Red Hat Linux 8
```
Comment (lukech) [inline test/evergreen.yml]: Suggest to use `RHEL 8.0`

File: test/evergreen.yml [hunk 2]
```diff
@@ -1266,6 +1266,74 @@ buildvariants:
...
+  run_on:
+  - rhel80-test
+  expansions:
+    # It's ugly, but we need the absolute path here, not the relative
```
Comment (lukech) [inline test/evergreen.yml]: Prefer to remove this ugly comment :)

File: test/evergreen.yml [hunk 3]
```diff
@@ -1266,6 +1266,37 @@ buildvariants:
...
+  run_on:
+  - rhel80-test
+  expansions:
+    test_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+    configure_env_vars: CC=/opt/mongodbtoolchain/v3/bin/gcc CXX=/opt/mongodbtoolchain/v3/bin/g++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+    make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
+  tasks:
+    - name: compile
+    - name: lang-python-test
```
Comment (lukech) [inline test/evergreen.yml]: This should have been covered by the next line `make-check-test`.

File: test/evergreen.yml [hunk 4]
```diff
@@ -1266,6 +1266,37 @@ buildvariants:
...
+    - name: make-check-test
+    - name: unit-test
+    - name: fops
```
Comment (lukech) [inline test/evergreen.yml]: Ditto

File: test/evergreen.yml [hunk 5]
```diff
@@ -1266,6 +1266,37 @@ buildvariants:
...
+    test_env_vars: PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+    configure_env_vars: CC=/opt/mongodbtoolchain/v3/bin/gcc CXX=/opt/mongodbtoolchain/v3/bin/g++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+    configure_python_setting: PYTHON=python3
+    make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
+    python_binary: python3
+  tasks:
+    - name: compile
+    - name: unit-test
+
```
Comment (lukech) [inline test/evergreen.yml]: As the ubuntu build variant has covered Python2 tests and Python2 is going to EOL support, I'd suggest we make Python3 the default and only Python testing for the new rhel80 build variant - i.e. merging this `rhel80-python3` build variant with `rhel80`.

**Comment by quchenhao** [comment]
retest this please

**Pull request merged**